src/editor/Gui.cpp

```diff
@@ -586,22 +623,81 @@ void Gui::drawMenuBar() {
 				if (map)
 				{
 					logf("Export wad: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), (map->name + ".wad").c_str());
-					ExportWad(map);
-					logf("Remove all embedded textures\n");
-					map->delete_embedded_textures();
-					if (map->ents.size())
+					if (ExportWad(map))
 					{
-						std::string wadstr = map->ents[0]->keyvalues["wad"];
-						if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+						logf("Remove all embedded textures\n");
+						map->delete_embedded_textures();
+						if (map->ents.size())
+						{
+							std::string wadstr = map->ents[0]->keyvalues["wad"];
+							if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+							{
+								map->ents[0]->keyvalues["wad"] += map->name + ".wad" + ";";
+							}
+						}
+					}
+				}
+			}
+
+			if (ImGui::MenuItem("Wavefront (.obj)", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					map->ExportToObjWIP(g_settings.gamedir + g_settings.workingdir);
+				}
+			}
+
+			if (ImGui::BeginMenu("Models")) {
+				for (int r = 0; r < app->mapRenderers.size(); r++)
+				{
+					if (!app->mapRenderers[r]->map)
+						continue;
+					if (ImGui::BeginMenu((std::string("Map ") + app->mapRenderers[r]->map->name + ".bsp").c_str())) {
+						Bsp* map = app->mapRenderers[r]->map;
+						if (map)
 						{
-							map->ents[0]->keyvalues["wad"] += map->name + ".wad" + ";";
+							for (int i = 0; i < map->modelCount; i++)
+							{
+								if (ImGui::MenuItem(("Export " + std::to_string(i) + " model (.bsp)").c_str(), NULL, app->pickInfo.modelIdx == i)) {
+									if (fileExists(map->path))
+									{
+										Bsp* tmpMap = new Bsp(map->path);
+										tmpMap->modelCount = 1;
+										tmpMap->models[0] = map->models[i];
+										STRUCTCOUNT removed = tmpMap->remove_unused_model_structures();
+
+										if (!removed.allZero())
+											removed.print_delete_stats(1);
+
+										Entity* tmpEnt = tmpMap->ents[0];
+										for (int i = 1; i < tmpMap->ents.size(); i++)
+											delete tmpMap->ents[i];
+										tmpMap->ents.clear();
+										tmpMap->ents.push_back(tmpEnt);
+										tmpMap->delete_unused_hulls();
+										tmpMap->update_ent_lump();
+										tmpMap->update_lump_pointers();
+										logf("Export model %d to %s\n", i, (g_settings.gamedir + g_settings.workingdir + "model" + std::to_string(i) + ".bsp").c_str());
+										tmpMap->write(g_settings.gamedir + g_settings.workingdir + "model" + std::to_string(i) + ".bsp");

```

Comment:
same comments here as above. Also this seems to mostly be duplicate code, so should move into its own method.